### DISCUSSION TITLE
Anaconda problem

### DISCUSSION THREAD

**Discussion by Naimas** [MightyBOBcnc/nixis#9] [Ideas]
I am quite interested in this , but I tried install Anaconda and miniconda but I only get errors over errors, is it possible to have it run otherwise in a non linux environment? Also I have other ideas / features that could be implemented like pressure zones and sea currents that would help shape the biomes better.

**Comment by MightyBOBcnc** [comment]
Hi there.  Although I'm developing in an Anaconda environment on Windows 10 the project shouldn't necessarily require it.  I just found that to be easier to get the environment set up.  A pyenv should also work although I haven't tried that myself.  

Most of the heavy lifting is done by:
```
meshzoo (for generating the planet icosphere)
numpy
numba (for not-garbage erosion/insolation speed; not strictly required but performance will be harmed without it)
scipy (for KD Tree)
numba-optimized opensimplex fork (included in Nixis repo) (for noise)
```

For visualizing the results:
```
pyvista
matplotlib (for color gradient stuff)
cmocean (for color gradient stuff)
```

For exporting:
```
pillow (exporting images)
meshio (exporting meshes)
```

Almost everything else in the (admittedly bloated) yaml file are the dependencies for these core packages.  So what you could try to do is install the core packages by hand (aside from the ones that come standard with python) in the traditional way and let them sort out their own dependencies.  I guess the main caveat is that I haven't had the ability to work on the project in a year (I **do** plan to return) so I haven't tested newer versions of those packages than what is listed in the yaml, or python versions newer than 3.8.10.  In theory nothing in the newer packages should be incompatible with Nixis itself but I have no idea about possible dependency conflicts between the big packages like numpy, numba, scipy, pyvista, and matplotlib.  I know, for example, that numba has been updating their minimum supported numpy version in more recent releases.

Yes I would definitely like to try approximated pressure zones, atmospheric circulation, and sea currents.

**Comment by Naimas** [comment]
What king of Climate models did you use for that kind of built?

**Reply by MightyBOBcnc** [reply]
There is no climate model yet.  Right now there is only a basic experiment with equilibrium temperature and insolation accumulation.  More effort is being dedicated to hydraulic erosion because without making functional erosion there's no point in doing the rest of the project.  In the future I'd like to try a basic 2 layer atmosphere model [(example)](https://biocycle.atmos.colostate.edu/shiny/2layer/#tab-4744-4) for energy budget, along with pressure and wind cells + Coriolis to tackle air circulation and distribution of atmospheric water vapor to locate areas of precipitation and rain shadows behind mountains.
